arrows/ffmpeg/ffmpeg_video_input.h

```diff
@@ -0,0 +1,92 @@
+/*ckwg +29
+ * Copyright 2018 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief \todo
+ */
+
+#ifndef KWIVER_ARROWS_FFMPEG_ffmpeg_video_input_H
+#define KWIVER_ARROWS_FFMPEG_ffmpeg_video_input_H
+
+#include <vital/algo/video_input.h>
+
+#include <arrows/ffmpeg/kwiver_algo_ffmpeg_export.h>
+
+
+namespace kwiver {
+namespace arrows {
+namespace ffmpeg {
+
+/// Video input using VXL vidl ffmpeg services.
+// ----------------------------------------------------------------
+/**
+ * This class implements a video input algorithm using the VXL vidl
+ * ffmpeg video services.
+ *
+ */
+class KWIVER_ALGO_FFMPEG_EXPORT ffmpeg_video_input
+  : public vital::algorithm_impl < ffmpeg_video_input, vital::algo::video_input >
+{
```

Comment:
Since this is derived from video_input it should support algorithm capabilities.